src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -191,6 +192,55 @@ public void requestHandlerTest() throws IOException {
 		assertEquals("Listener failed to accept connections exactly once from sender in http mode.", 1, handlerExecuted.get());
 	}
 	
+	@Test
+	public void parseRequestPathAndQueryTest() throws IOException, URISyntaxException {
+		CompletableFuture<Void> requestReceived = new CompletableFuture<Void>();
+		String extraPath = "/extraPath/extraPath2";
+		String extraQuery = "queryKey=queryValue&queryKey2=queryValue;"; // mixing in a special char just for fun
+		
+		listener.setRequestHandler((context) -> {
+			try {
+				assertNotNull("Listener should have received a valid http context.", context);
+				assertNotNull("Listener should have received a valid http request.", context.getRequest());
+				
+				URI requestUri = context.getRequest().getUri();
+				assertNotNull("Listener should have a URI from request", requestUri);
+				assertEquals(
+						"The path wasn't the expected value", 
+						"/bailiu-relay-hc/bailiu-relay-hc" + extraPath, 
+						requestUri.getPath()
+				);
+				assertEquals("The query wasn't the expected value", extraQuery, requestUri.getQuery());
+				
+				requestReceived.complete(null);
+			} catch (Throwable ex) {
+				requestReceived.completeExceptionally(ex);
```

Comment:
I love this. At first I wondered how asserts in this handler would get reported.